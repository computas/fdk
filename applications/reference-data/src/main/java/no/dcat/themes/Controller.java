package no.dcat.themes;

import no.dcat.shared.DataTheme;
import no.dcat.shared.HelpText;
import no.dcat.shared.LocationUri;
import no.dcat.shared.SkosCode;
import no.dcat.shared.Subject;
import no.dcat.shared.Types;
import no.dcat.themes.service.CodesService;
import no.dcat.themes.service.HelpTextService;
import no.dcat.themes.service.SubjectsService;
import no.dcat.themes.service.ThemesService;
import org.apache.jena.shared.NotFoundException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Scope;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import java.net.MalformedURLException;
import java.util.List;

@RestController
@Scope("thread")

public class Controller {

    @Autowired
    private CodesService codesService;

    @Autowired
    private HelpTextService helpTextService;

    @Autowired
    private ThemesService themesService;

    @Autowired
    private SubjectsService subjectsService;

    static private final Logger logger = LoggerFactory.getLogger(Controller.class);


    @CrossOrigin
    @RequestMapping(value = "/codes", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public List<String> codeTypes() {

        return codesService.listCodes();

    }


    @CrossOrigin
    @RequestMapping(value = "/codes/{type}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public List<SkosCode> codes(@PathVariable(name = "type") String type) {

        return codesService.getCodes(Types.valueOf(type));

    }


    @CrossOrigin
    @RequestMapping(value = "/themes", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public List<DataTheme> themes() {

        return themesService.getThemes();

    }

    @CrossOrigin
    @RequestMapping(value = "/helptexts", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public List<HelpText> helpTexts() {

        return helpTextService.getHelpTexts();

    }

    @CrossOrigin
    @RequestMapping(value = "/helptexts/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    @ExceptionHandler(NotFoundException.class)
    @ResponseStatus(HttpStatus.NOT_FOUND)
    public HelpText helpTexts(@PathVariable(name = "id") String id) {

        return helpTextService.getHelpTexts(id);

    }

    @PreAuthorize("hasAuthority('INTERNAL_CALL')")
    @CrossOrigin
    @RequestMapping(value = "/locations", consumes = MediaType.APPLICATION_JSON_UTF8_VALUE, method = RequestMethod.POST)
    public SkosCode putLocation(@RequestBody LocationUri resource) throws MalformedURLException {
        logger.info("register new location: {}", resource.getUri());
        try {
            return codesService.addLocation(resource.getUri());
        }catch (Exception e){
            logger.error("Unable to find location with URI <{}>. Reason {}",resource.getUri(), e.getMessage());
            throw e;
        }
    }

    @PreAuthorize("hasAuthority('INTERNAL_CALL')")
    @CrossOrigin
    @RequestMapping(value = "/subjects",  method = RequestMethod.GET)
    public Subject getRemoteResourceForSubject(@RequestParam String uri) throws MalformedURLException {
        logger.info("Request for subject with uri <{}>", uri);
        try {
            Subject subject = subjectsService.addSubject(uri);
            logger.info("Return subject: {}", subject);
            return subject;
        }catch (Exception e){
            logger.error("Unable to find subject with URI <{}>. Reason {}",uri, e.getLocalizedMessage());
            throw e;
        }
    }


}
