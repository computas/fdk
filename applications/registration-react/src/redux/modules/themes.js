import {
  THEMES_REQUEST,
  THEMES_SUCCESS,
  THEMES_FAILURE
} from '../../constants/ActionTypes';

export default function themes(
  state = { isFetchingThemes: false, themesItems: null },
  action
) {
  switch (action.type) {
    case THEMES_REQUEST:
      return {
        ...state,
        isFetchingThemes: true
      };
    case THEMES_SUCCESS:
      return {
        ...state,
        isFetchingThemes: false,
        themesItems: action.payload
      };
    case THEMES_FAILURE:
      return {
        ...state,
        isFetchingThemes: false,
        themesItems: null
      };
    default:
      return state;
  }
}
