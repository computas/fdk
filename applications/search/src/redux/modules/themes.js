import { fetchActions } from '../fetchActions';

export const THEMES_REQUEST = 'THEMES_REQUEST';
export const THEMES_SUCCESS = 'THEMES_SUCCESS';
export const THEMES_FAILURE = 'THEMES_FAILURE';

export function fetchThemesIfNeededAction() {
  return (dispatch, getState) => {
    if (!getState().themes.isFetching) {
      dispatch(
        fetchActions('/reference-data/themes', [
          THEMES_REQUEST,
          THEMES_SUCCESS,
          THEMES_FAILURE
        ])
      );
    }
  };
}

const initialState = { isFetching: false, themesItems: null };

export function themesReducer(state = initialState, action) {
  switch (action.type) {
    case THEMES_REQUEST:
      return {
        ...state,
        isFetching: true
      };
    case THEMES_SUCCESS: {
      const objFromArray = action.payload.reduce((accumulator, current) => {
        accumulator[current.code] = current;
        return accumulator;
      }, {});
      return {
        ...state,
        isFetching: false,
        themesItems: objFromArray
      };
    }
    case THEMES_FAILURE: {
      return {
        ...state,
        isFetching: false,
        themesItems: null
      };
    }
    default:
      return state;
  }
}
