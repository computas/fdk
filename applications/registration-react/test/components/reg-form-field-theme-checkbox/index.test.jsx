import React from 'react';
import { shallow } from 'enzyme';
import CheckboxFieldTheme from '../../../src/components/reg-form-field-theme-checkbox';
import themes from '../../fixtures/themes';

let defaultProps;
let wrapper;

beforeEach(() => {
  const { themesItems } = themes;
  defaultProps = {
    input: { name: 'theme', value: [] },
    label: 'label',
    themesItems
  };
  wrapper = shallow(<CheckboxFieldTheme {...defaultProps} />);
});

test('should render CheckboxFieldTheme correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
