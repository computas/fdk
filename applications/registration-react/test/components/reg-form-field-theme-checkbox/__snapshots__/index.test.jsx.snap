// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CheckboxFieldTheme correctly 1`] = `
<div>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/AGRI"
    key="0"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/AGRI"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/AGRI"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Jordbruk, fiskeri, skogbruk og mat
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/ECON"
    key="1"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/ECON"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/ECON"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Økonomi og finans
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/EDUC"
    key="2"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/EDUC"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/EDUC"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Utdanning, kultur og sport
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/ENER"
    key="3"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/ENER"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/ENER"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Energi
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/ENVI"
    key="4"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/ENVI"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/ENVI"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Miljø
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/GOVE"
    key="5"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/GOVE"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/GOVE"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Forvaltning og offentlig sektor
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/HEAL"
    key="6"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/HEAL"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/HEAL"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Helse
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/INTR"
    key="7"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/INTR"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/INTR"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Internasjonale temaer
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/JUST"
    key="8"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/JUST"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/JUST"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Justis, rettssystem og allmenn sikkerhet
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/REGI"
    key="9"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/REGI"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/REGI"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Regioner og byer
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/SOCI"
    key="10"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/SOCI"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/SOCI"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Befolkning og samfunn
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/TECH"
    key="11"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/TECH"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/TECH"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Vitenskap og teknologi
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="http://publications.europa.eu/resource/authority/data-theme/TRAN"
    key="12"
  >
    <input
      checked=""
      id="http://publications.europa.eu/resource/authority/data-theme/TRAN"
      name="themes"
      onChange={[Function]}
      type="checkbox"
      value="http://publications.europa.eu/resource/authority/data-theme/TRAN"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Transport
    </span>
  </label>
</div>
`;
